<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NewDataTypeForm</class>
 <widget class="QWidget" name="NewDataTypeForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>960</width>
    <height>662</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="focusPolicy">
   <enum>Qt::TabFocus</enum>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_2">
   <item>
    <widget class="QFrame" name="frameForUnlockButton">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="QToolButton" name="btnLock">
        <property name="text">
         <string/>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="popupMode">
         <enum>QToolButton::InstantPopup</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="btnUndelete">
        <property name="toolTip">
         <string>This object was deleted in a Patch. Click this button to Undelete it.</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="dbm.qrc">
          <normaloff>:/images/actions/images/small/gnome_undelete.png</normaloff>:/images/actions/images/small/gnome_undelete.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
        <property name="checkable">
         <bool>false</bool>
        </property>
        <property name="popupMode">
         <enum>QToolButton::InstantPopup</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="grpContent">
     <property name="title">
      <string/>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="tabWidgetPage1">
         <attribute name="title">
          <string>Basic properties</string>
         </attribute>
         <layout class="QFormLayout" name="formLayout">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Name</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="txtDTName"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Type</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="cmbDTType">
            <property name="currentIndex">
             <number>0</number>
            </property>
            <item>
             <property name="text">
              <string>Text/String</string>
             </property>
             <property name="icon">
              <iconset resource="dbm.qrc">
               <normaloff>:/images/actions/images/small/datatypes_text.png</normaloff>:/images/actions/images/small/datatypes_text.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Numeric</string>
             </property>
             <property name="icon">
              <iconset resource="dbm.qrc">
               <normaloff>:/images/actions/images/small/datatypes_numeric.png</normaloff>:/images/actions/images/small/datatypes_numeric.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Boolean</string>
             </property>
             <property name="icon">
              <iconset resource="dbm.qrc">
               <normaloff>:/images/actions/images/small/datatypes_bool.png</normaloff>:/images/actions/images/small/datatypes_bool.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Date/Time</string>
             </property>
             <property name="icon">
              <iconset resource="dbm.qrc">
               <normaloff>:/images/actions/images/small/datatypes_datetime.png</normaloff>:/images/actions/images/small/datatypes_datetime.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Blob/Binary</string>
             </property>
             <property name="icon">
              <iconset resource="dbm.qrc">
               <normaloff>:/images/actions/images/small/datatypes_blob.png</normaloff>:/images/actions/images/small/datatypes_blob.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Misc</string>
             </property>
             <property name="icon">
              <iconset resource="dbm.qrc">
               <normaloff>:/images/actions/images/small/datatypes_misc.png</normaloff>:/images/actions/images/small/datatypes_misc.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Spatial</string>
             </property>
             <property name="icon">
              <iconset resource="dbm.qrc">
               <normaloff>:/images/actions/images/small/datatypes_spatial.png</normaloff>:/images/actions/images/small/datatypes_spatial.png</iconset>
             </property>
            </item>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>SQL</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QComboBox" name="cmbDTSQLType"/>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="lblWidth">
            <property name="text">
             <string>Width</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="txtWidth"/>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="lblValues">
            <property name="text">
             <string>Values</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <layout class="QGridLayout" name="layoutEnums">
            <item row="0" column="0">
             <widget class="QListWidget" name="lstEnumValues"/>
            </item>
            <item row="0" column="1">
             <widget class="QToolButton" name="btnRemoveEnumValue">
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset resource="dbm.qrc">
                <normaloff>:/images/actions/images/small/remove.png</normaloff>:/images/actions/images/small/remove.png</iconset>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QToolButton" name="btnAddEnumValue">
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset resource="dbm.qrc">
                <normaloff>:/images/actions/images/small/add.png</normaloff>:/images/actions/images/small/add.png</iconset>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLineEdit" name="txtEnumCurrentValue"/>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tabWidgetPage2">
         <attribute name="title">
          <string>Advanced</string>
         </attribute>
         <layout class="QFormLayout" name="formLayout_2">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
          </property>
          <item row="1" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Default Value</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QComboBox" name="cmbEnumItems"/>
            </item>
            <item>
             <widget class="QLineEdit" name="txtDefaultValue"/>
            </item>
            <item>
             <widget class="QCheckBox" name="chkNullIsDefault">
              <property name="text">
               <string>Null</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="lblUnsigned">
            <property name="text">
             <string>Sign</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QCheckBox" name="chkUnsigned">
            <property name="text">
             <string>Unsigned</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>NULL value allowed</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QCheckBox" name="chkCanBeNull">
            <property name="text">
             <string>Yes</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tabWidgetPage3">
         <attribute name="title">
          <string>Description</string>
         </attribute>
         <layout class="QFormLayout" name="formLayout_3">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Description</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QTextEdit" name="txtDescription">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_6">
        <property name="title">
         <string/>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="btnSave">
           <property name="text">
            <string>Save</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="btnHelp">
           <property name="text">
            <string>Help</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="dbm.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>cmbDTType</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>basicDTselected(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmbDTSQLType</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onSqlTypeSelected(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>128</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnAddEnumValue</sender>
   <signal>clicked()</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onAddMiscValue()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>363</x>
     <y>395</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnRemoveEnumValue</sender>
   <signal>clicked()</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onRemoveSelectedMiscValue()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>363</x>
     <y>277</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkNullIsDefault</sender>
   <signal>clicked()</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onNullClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>272</x>
     <y>138</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkCanBeNull</sender>
   <signal>clicked()</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onCanBeNullClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>272</x>
     <y>209</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnHelp</sender>
   <signal>clicked()</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onHelp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>238</x>
     <y>552</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnSave</sender>
   <signal>clicked()</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>81</x>
     <y>612</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnLock</sender>
   <signal>toggled(bool)</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onLockUnlock(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>37</x>
     <y>36</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnUndelete</sender>
   <signal>clicked()</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onUndelete()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>38</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>basicDTselected(QString)</slot>
  <slot>onSave()</slot>
  <slot>onReset()</slot>
  <slot>onSqlTypeSelected(QString)</slot>
  <slot>onAddMiscValue()</slot>
  <slot>onRemoveSelectedMiscValue()</slot>
  <slot>onCharacterSetCmbChanged(int)</slot>
  <slot>onNullClicked()</slot>
  <slot>onCanBeNullClicked()</slot>
  <slot>onHelp()</slot>
  <slot>onLockUnlock(bool)</slot>
  <slot>onUndelete()</slot>
 </slots>
</ui>
